src/compas/files/_BaseReader.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+    def __init__(self, location):
+        self.is_URL = False
```

Comment:
this should be all lowercase:

```suggestion
        self.is_url = False
```